### PR TITLE
Swift enum names capitalization follows Swift convention.

### PR DESCRIPTION
**Pull request by tomekc**
Here is a solution for #1420 that I have reported. I introduced unit test for SwiftCodegen which documents the behavior by the way.

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SwiftCodegen.java [hunk 1]
```diff
@@ -250,6 +251,16 @@ public CodegenProperty fromProperty(String name, Property p) {
...
   }
 
+    public String toSwiftyEnumName(String value) {
+        // Prevent from breaking properly cased identifier
+        if (value.matches("[A-Z][a-z0-9]+[a-zA-Z0-9]*")) {
+            return value;
+        }
+        char[] separators = {'-', '_', ' '};
+        return WordUtils.capitalizeFully(StringUtils.lowerCase(value), separators).replaceAll("[-_ ]", "");
+    }
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SwiftCodegen.java]: There are 2 methods that can help you sanitize and camelize the Swift Enum name:

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java#L1662

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java#L1549

Here is how CSharp uses these 2 methods for variable naming.

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/CSharpClientCodegen.java#L148

**Comment by tomekc** [comment]
I put camelize() in place of my implementation and it failed miserably. For example it doesn't convert all-caps names like ENTRY_NAME or ENTRY.

By the way, CSharp generator ignores Enums and generates plain property of given type.

So I would really like to keep my implementation.

**Comment by wing328** [comment]
@tomekc ok. Just wanted to let you know there are utility functions that may help in your case.

**Comment by wing328** [comment]
@tomekc and thanks for writing test cases for the change. Later we might move the test file to the "swift" folder instead.

**Pull request merged**